etcdserver/api/membership/cluster_test.go

```diff
@@ -859,3 +880,221 @@ func TestIsReadyToRemoveVotingMember(t *testing.T) {
 		}
 	}
 }
+
+func TestMustDetectDowngrade(t *testing.T) {
+	lv := semver.Must(semver.NewVersion(version.Version))
+	lv = &semver.Version{Major: lv.Major, Minor: lv.Minor}
+	oneMinorHigher := &semver.Version{Major: lv.Major, Minor: lv.Minor + 1}
+	oneMinorLower := &semver.Version{Major: lv.Major, Minor: lv.Minor - 1}
+	downgradeEnabledHigherVersion := &Downgrade{Enabled: true, TargetVersion: oneMinorHigher}
+	downgradeEnabledEqualVersion := &Downgrade{Enabled: true, TargetVersion: lv}
+	downgradeEnabledLowerVersion := &Downgrade{Enabled: true, TargetVersion: oneMinorLower}
+	downgradeDisabled := &Downgrade{Enabled: false}
+
+	tests := []struct {
+		currentVersion *semver.Version
```

Comment:
Let's add test name here so we know which one fails, something like "Downgrade enabled, target version match, should succeed."

Reply:
Added.